alvr/adb/src/lib.rs

```diff
@@ -1,395 +1,78 @@
+        let adb_path = commands::require_adb(layout, download_progress_callback)?;
 
-pub fn get_process_id(
-    adb_path: &str,
-    device_serial: &str,
-    process_name: &str,
-) -> Result<Option<usize>> {
-    let output = get_command(
-        adb_path,
-        &["-s", device_serial, "shell", "pidof", process_name],
-    )
-    .output()
-    .context(format!("Failed to get ID of process {process_name}"))?;
-    let text = String::from_utf8_lossy(&output.stdout).trim().to_owned();
-    if text.is_empty() {
-        return Ok(None);
+        Ok(Self { adb_path })
     }
-    let process_id = text
-        .parse::<usize>()
-        .context("Failed to parse process ID")?;
 
-    Ok(Some(process_id))
-}
+    pub fn setup(&self, control_port: u16, stream_port: u16) -> Result<WiredConnectionStatus> {
+        let Some(device_serial) = commands::list_devices(&self.adb_path)?
+            .into_iter()
+            .filter_map(|d| d.serial)
+            .find(|s| !s.starts_with("127.0.0.1"))
+        else {
+            return Ok(WiredConnectionStatus::NotReady(
+                "No wired devices found".to_owned(),
+            ));
+        };
+
+        let ports = HashSet::from([control_port, stream_port]);
+        let forwarded_ports: HashSet<u16> =
+            commands::list_forwarded_ports(&self.adb_path, &device_serial)?
+                .into_iter()
+                .map(|f| f.local)
+                .collect();
+        let missing_ports = ports.difference(&forwarded_ports);
+        for port in missing_ports {
+            commands::forward_port(&self.adb_path, &device_serial, *port)?;
+            dbg_connection!(
+                "setup_wired_connection: Forwarded port {port} of device {device_serial}"
+            );
+        }
 
-pub fn is_activity_resumed(
-    adb_path: &str,
-    device_serial: &str,
-    activity_name: &str,
-) -> Result<bool> {
-    let output = get_command(
-        adb_path,
-        &[
-            "-s",
-            device_serial,
-            "shell",
-            "dumpsys",
-            "activity",
-            activity_name,
-        ],
-    )
-    .output()
-    .context(format!("Failed to get state of activity {activity_name}"))?;
-    let text = String::from_utf8_lossy(&output.stdout);
-    if let Some(line) = text
-        .lines()
-        .map(|l| l.trim())
-        .find(|l| l.contains("mResumed"))
-    {
-        let (entry, _) = line
-            .split_once(' ')
-            .ok_or(anyhow!("Failed to split resumed state line"))?;
-        let (_, value) = entry
-            .split_once('=')
-            .ok_or(anyhow!("Failed to split resumed state entry"))?;
-        match value {
-            "true" => Ok(true),
-            "false" => Ok(false),
-            _ => Err(anyhow!("Failed to parse resumed state value"))?,
+        #[cfg(debug_assertions)]
+        let process_name = "alvr.client.dev";
+        #[cfg(not(debug_assertions))]
+        let process_name = "alvr.client.stable";
```

Comment:
Also potentially out of the scope of this pr, but I have a lot of different client names for testing and a way to override this in the settings would be pretty practical

Reply:
oh yeah I agree. or maybe provide some fallbacks